scripts/check_deployment_secrets.py

```diff
@@ -0,0 +1,156 @@
+    `project_id` should be `human-cell-atlas-travis-test`
+    `type` should be `service_account`
+    `client_email` should be the user account returned from the terraform output "service_account".
+                   For example: dev should be `travis-test@human-cell-atlas-travis-test.iam.gserviceaccount.com`.
+"""
+import subprocess
+import os
+import sys
+import json
+import copy
+
+
+class SecretsChecker(object):
+    def __init__(self, stage):
+        self.stage = stage
+        self.stages = {'dev': 'environment',
+                       'integration': 'enviroment.integration',
+                       'staging': 'environment.staging',
+                       'prod': 'environment.prod'}
+
+        self.missing_secrets = []
+        self.malformed_secrets = []
+        self.incomplete_secrets = []
+        self.error_message = f'\n\n' \
+                             f'Deploying to {self.stage.upper()} could not be completed.\n' \
+                             f'It looks like an AWS secret has an unexpected value.\n' \
+                             f'Please do not change AWS secrets for releases.\n'
+
+        if self.stage not in self.stages:
+            print(f'Custom stage "{self.stage}" provided.  Secret checking skipped.')
+            return
+
+        self.stage_env = copy.deepcopy(os.environ)
+        self.stage_env = self.get_stage_env(self.stages[self.stage])
+        self.service_account = self.fetch_terraform_output("service_account", "gcp_service_account").strip()
+
+        self.email = [f'{self.service_account}@human-cell-atlas-travis-test.iam.gserviceaccount.com']
+        self.project = ['human-cell-atlas-travis-test']
+        self.type = ['service_account']
+        self.auth_uri = ['https://auth.data.humancellatlas.org/authorize',
+                         'https://auth.dev.data.humancellatlas.org/authorize']
+        self.token_uri = ['https://auth.data.humancellatlas.org/oauth/token',
+                          'https://auth.dev.data.humancellatlas.org/oauth/token']
+
+        self.app_secret_name = os.environ['GOOGLE_APPLICATION_SECRETS_SECRETS_NAME']
+        self.gcp_cred_secret_name = os.environ['GOOGLE_APPLICATION_CREDENTIALS_SECRETS_NAME']
+        self.app_secret = self.fetch_secret(self.app_secret_name)
+        self.gcp_cred_secret = self.fetch_secret(self.gcp_cred_secret_name)
+
+    def run_cmd(self, cmd, cwd=os.getcwd(), shell=True):
+        p = subprocess.Popen(cmd,
+                             shell=shell,
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE,
+                             cwd=cwd,
+                             env=self.stage_env)
+        stdout, _ = p.communicate()
+        return stdout.decode('utf-8')
+
+    def get_stage_env(self, env_file):
+        dump = '/usr/bin/python -c "import os, json; print(json.dumps(dict(os.environ)))"'
+        cmd = ['/bin/bash', '-c', f'source {env_file} && {dump}']
```

Comment:
may want to have `#!/usr/bin/env bash` to avoid breaks if `bash` is not in the `/bin/`. Not sure how applicable this is to our user-base.

Reply:
The equivalent to the shebang in a straight cli command would be just using `bash` instead of `/bin/bash` since they both pull the first `bash` command off of the top of the available paths.  It's not a bad point though.  I'll change them from absolute paths.